code/modules/vending/_vending.dm

```diff
@@ -444,6 +454,15 @@ GLOBAL_LIST_EMPTY(vending_products)
 				flick(icon_deny,src)
 				vend_ready = 1
 				return
+			else if(age_restrictions && R.age_restricted && (!C.registered_age || C.registered_age < AGE_MINOR))
```

Comment:
You haven't defined R in this context.  It's defined in a for-loop up on line 403 in a different function.
The information you have about the product is in variable N (def. L439), which is a string containing the product name.  You need to loop through the vending products looking for the one with name N and check its age restricted separately.
Example:
```dm
else if(age_restrictions && (!C.registered_age || C.registered_age < AGE_MINOR))
    var/product_is_age_restricted = FALSE
    for (var/datum/data/vending_product/R in display_records)
        // Assumption: vending_product datum name always equals product name
        // (true in this case)
        if (R.name == N && R.age_restricted)
            product_is_age_restricted = TRUE
            break

    if (product_is_age_restricted)
        // Your code here
```